src/transformers/modeling_tf_utils.py

```diff
@@ -584,59 +614,121 @@ def __init__(self, config, *inputs, **kwargs):
         self.config = config
         self.name_or_path = config.name_or_path
 
-    def get_input_embeddings(self) -> tf.keras.layers.Layer:
+    def get_input_embeddings(self) -> tf.Variable:
         """
-        Returns the model's input embeddings.
+        Returns the model's input embeddings layer.
 
         Returns:
-            :obj:`tf.keras.layers.Layer`: A torch module mapping vocabulary to hidden states.
+            :obj:`tf.Variable`: The embeddings layer mapping vocabulary to hidden states.
         """
-        base_model = getattr(self, self.base_model_prefix, self)
+        main_layer = getattr(self, self.base_model_prefix, self)
 
-        if base_model is not self:
-            return base_model.get_input_embeddings()
+        if main_layer is not self:
+            try:
+                return main_layer.get_input_embeddings()
+            except AttributeError:
+                self(self.dummy_inputs)
```

Comment:
What is the use case of this except statement -> can you clarify a bit why we need this functionality?

Reply:
Because we need to be sure that the model is entirely built and not partially.